doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+output (to which a bias vector is usually added before passing the result
+through a nonlinearity). This is applicable to any type of input, be it an
+image, a sound clip or an unordered collection of features: whatever their
+dimensionality, their representation can always be flattened into a vector
+before the transformation.
+
+Images, sound clips and many other similar kinds of data have an intrinsic
+structure. More formally, they share these important properties:
+
+* They are stored as multi-dimensional arrays.
+* They feature one or more axes for which ordering matters (e.g., width and
+  height axes for an image, time axis for a sound clip).
+* One axis, called the channel axis, is used to access different views of the
+  data (e.g., the red, green and blue channels of a color image, or the left
+  and right channels of a stereo audio track).
+
+These properties are not exploited when an affine transformation is applied; in
+fact, all the axes are treated in the same way and the topological information
+is not taken into account. Still, taking advantage of the implicit structure of
+the data may prove very handy in solving some tasks, like computer vision and
+speech recognition, and in these cases it would be best to preserve it. This is
+where discrete convolutions come into play.
+
+A discrete convolution is a linear transformation that preserves this notion of
+ordering. It is sparse (only a few input units contribute to a given output
+unit) and reuses parameters (the same weights are applied to multiple locations
+in the input).
+
+Here is an example of a discrete convolution:
+
+.. figure:: conv_arithmetic_figures/numerical_no_padding_no_strides.gif
+
+The light blue grid is called the *input feature map*. To keep the drawing
+simple, a single input feature map is represented, but it is not uncommon to
+have multiple feature maps stacked one onto another. (An example of this is
+what was referred to earlier as *channels* for images and sound clips.)
+A *kernel* (shaded area) of value
+
+.. math::
+
+    \begin{pmatrix}
+    0 & 1 & 2 \\
+    2 & 2 & 0 \\
+    0 & 1 & 2
+    \end{pmatrix}
+
+slides across the input feature map. At each location, the product between each
+element of the kernel and the input element it overlaps is computed and the
+results are summed up to obtain the output in the current location. The
+procedure can be repeated using different kernels to form as many output feature
+maps as desired. The final outputs of this procedure are called *output feature
+maps*.
+
+.. note::
+
+    While there is a distinction between convolution and cross-correlation from
+    a signal processing perspective, the two become interchangeable when the
+    kernel is learned. For the sake of simplicity and to stay consistent with
+    most of the machine learning literature, the term *convolution* will be
+    used in this tutorial.
+
+If there are multiple input feature maps, the kernel will have to be
```

Comment:
"each kernel"

I think it's more clear to go this way, even if the figures just represent the single feature map case.